include/SW_datastructs.h

```diff
@@ -0,0 +1,1055 @@
+/*                 Output struct/enums                 */
+/* --------------------------------------------------- */
+
+typedef enum {
+	eSW_Off, eSW_Sum, eSW_Avg, eSW_Fnl
+} OutSum;
+
+/* these are the code analog of the above */
+/* see also key2str[] in Output.c */
+/* take note of boundary conditions in ForEach...() loops below */
+typedef enum {
+	eSW_NoKey = -1,
+	/* weather/atmospheric quantities */
+	eSW_AllWthr, /* includes all weather vars */
+	eSW_Temp,
+	eSW_Precip,
+	eSW_SoilInf,
+	eSW_Runoff,
+	/* soil related water quantities */
+	eSW_AllH2O,
+	eSW_VWCBulk,
+	eSW_VWCMatric,
+	eSW_SWCBulk,
+	eSW_SWABulk,
+	eSW_SWAMatric,
+	eSW_SWA,
+	eSW_SWPMatric,
+	eSW_SurfaceWater,
+	eSW_Transp,
+	eSW_EvapSoil,
+	eSW_EvapSurface,
+	eSW_Interception,
+	eSW_LyrDrain,
+	eSW_HydRed,
+	eSW_ET,
+	eSW_AET,
+	eSW_PET, /* really belongs in wth, but for historical reasons we'll keep it here */
+	eSW_WetDays,
+	eSW_SnowPack,
+	eSW_DeepSWC,
+	eSW_SoilTemp,
+    eSW_Frozen,
+	/* vegetation quantities */
+	eSW_AllVeg,
+	eSW_Estab,
+	// vegetation other */
+	eSW_CO2Effects,
+	eSW_Biomass,
+	eSW_LastKey /* make sure this is the last one */
+} OutKey;
+
+typedef struct {
+	OutKey mykey;
+	ObjType myobj;
+	OutSum sumtype;
+	Bool
+		use,		// TRUE if output is requested
+		has_sl;	// TRUE if output key/type produces output for each soil layer
+	TimeInt
+		first, last, 			/* first/last doy of current year, i.e., updated for each year */
+		first_orig, last_orig; /* first/last doy that were originally requested */
+
+	#if defined(RSOILWAT)
+	char *outfile; /* name of output */ //could probably be removed
+	#endif
+} SW_OUTPUT;
+
+typedef struct {
+	TimeInt tOffset; /* 1 or 0 means we're writing previous or current period */
+
+
+	// Variables describing output periods:
+	/** `timeSteps` is the array that keeps track of the output time periods that
+		are required for `text` and/or `array`-based output for each output key. */
+	OutPeriod timeSteps[SW_OUTNKEYS][SW_OUTNPERIODS];
+
+	/** The number of different time steps/periods that are used/requested
+			Note: Under STEPWAT2, this may be larger than the sum of `use_OutPeriod`
+				because it also incorporates information from `timeSteps_SXW`. */
+	IntUS used_OUTNPERIODS;
+
+	/** TRUE if time step/period is active for any output key. */
+	Bool use_OutPeriod[SW_OUTNPERIODS];
+
+
+	// Variables describing size and names of output
+	/** names of output columns for each output key; number is an expensive guess */
+	char *colnames_OUT[SW_OUTNKEYS][5 * NVEGTYPES + MAX_LAYERS];
+
+	/** number of output columns for each output key */
+	IntUS ncol_OUT[SW_OUTNKEYS];
+
+		/** \brief A 2-dim array of pointers to output arrays.
+
+	\note This should be initialized to NULL because they are defined globally
+		and thus have `static storage duration`.
+
+	The variable p_OUT used by rSOILWAT2 for output and by STEPWAT2 for
+	mean aggregation.
+	*/
+	RealD *p_OUT[SW_OUTNKEYS][SW_OUTNPERIODS];
```

Comment:
Elements previously in "SW_Output_outarray" should be wrapped by `if defined(SW_OUTARRAY) ...` because the entire file "SW_Output_outarray.c" was only compiled by rSOILWAT2 or STEPWAT2 but not SOILWAT2 (see makefile and Makevars).

This applies for instance to `*p_OUT`, `nrow_OUT`, etc.